notification/app/notification/services/AzureRawPushConverter.scala

```diff
@@ -0,0 +1,76 @@
+package notification.services
+
+import java.net.URL
+
+import azure.{AzureRawPush, Tag}
+import models.Link.{External, Internal}
+import models._
+import notification.models.azure
+import play.api.libs.json.Json
+import models.JsonUtils._
+
+class AzureRawPushConverter(conf: Configuration) {
+  def toAzureRawPush(push: Push): AzureRawPush = {
+    val azureNotification = push.notification match {
+      case bnn: BreakingNewsNotification => toAzureBreakingNews(bnn)
+      case cn: ContentNotification => toContent(cn)
+      case gan: GoalAlertNotification => toGoalAlert(gan)
+    }
+    AzureRawPush(
+      body = Json.stringify(Json.toJson(azureNotification)),
+      tags = tags(push.destination)
+    )
+  }
+
+  private def tags(destination: Either[Topic, UserId]) = destination match {
+    case Left(topic: Topic) => Some(Set(Tag.fromTopic(topic)))
+    case Right(user: UserId) => Some(Set(Tag.fromUserId(user)))
+  }
+
+  private def toUrl(link: Link): URL = link match {
```

Comment:
to avoid pattern matching here it could be defined on trait and case classes of Link